packages/hashi/src/storage.js

```diff
@@ -0,0 +1,127 @@
+                this.sessionData = message.params['data'];
+                this.sessionKeys = Object.keys(this.sessionData);
+                this.dataReceived = true;
+                resolve(this);
+              }
+            }
+            parentMessageReceived = parentMessageReceived.bind(this);
+
+            function timedOut() {
+                reject("Load timed out.");
+            }
+
+            window.addEventListener('message', parentMessageReceived);
+            this._sendMessage('hashiInitialized', {});
+            this.timer = setTimeout(timedOut, timeout);
+        });
+
+        this.initMessageSent = true;
+      }
+      return initDataPromise;
+    }
+
+    get length() {
+        return this.sessionKeys.length;
+    }
+
+    key(index) {
+        return this.sessionKeys[index];
+    }
+
+    getItem(keyName) {
+        // TODO: Evaluate the performance implications of using postMessage and waiting for value
+        return this.sessionData[keyName];
+    }
+
+    setItem(keyName, value) {
+        console.log("setItem called");
+        if (this.sessionKeys.indexOf(keyName) === -1) {
+            this.sessionKeys.push(keyName);
+        }
+        this.sessionData[keyName] = value
+        this._sendMessage('stateUpdated', this.sessionData);
+    }
+
+    removeItem(keyName) {
+        delete this.sessionData[keyName];
+
+        var keyIndex = this.sessionKeys.indexOf(keyName);
+        if (keyIndex != -1) {
+            this.sessionKeys.splice(keyIndex, 1);
+        }
+        this._sendMessage('stateUpdated', this.sessionData);
+    }
+
+    clear() {
+        this.sessionKeys = [];
+        this.sessionData = {};
+
+        this._sendMessage('stateUpdated', this.sessionData);
+    }
+}
+
+export const getLocalStorage = () => {
+    // even referencing localStorage can throw a SecurityError, so don't assign to localStorage by default here.
+    let storage = null;
+    try {
+        // while some browsers throw immediately when trying to access localStorage, others only throw when
+        // you try to write to it, so we use a write test as the canonical way of testing if it works
+        const key = "testing_that_local_storage_is_allowed_using_a_key_that_would_never_actually_be_used";
+        localStorage.setItem(key, "It works!");
+        localStorage.removeItem(key);
+        storage = localStorage;
+        // we need to make this a promise as well for API consistency since we need a promise for Hashi storage.
+        const storagePromise = new Promise((resolve, reject) => {resolve(storage)});
+        return storagePromise;
+    } catch (e) {
+        if (e.name === 'SecurityError') {
+            console.log("Running in sandboxed iFrame, setting localStorage to Hashi.Storage instance.");
+            storage = new Storage();
+            return storage.initData();
+        } else {
+            throw e;
+        }
+    }
+    return storage;
+}
```

Comment:
looks like `prettier` isn't being run on this file. Recommend setting that up to prevent a big ugly diff down the line